editor/src/clj/internal/transaction.clj

```diff
@@ -723,17 +730,23 @@
                     (in/type-name input-nodetype) target-label))
     (assert-schema-type-compatible source-id source-label output-nodetype output-valtype target-id target-label input-nodetype input-valtype)))
 
-(defn- ctx-connect [ctx source-id source-label target-id target-label]
-  (if-let [source (gt/node-by-id-at (:basis ctx) source-id)] ; nil if source node was deleted in this transaction
-    (if-let [target (gt/node-by-id-at (:basis ctx) target-id)] ; nil if target node was deleted in this transaction
+(defn- ctx-connect [{:keys [basis] :as ctx} source-id source-label target-id target-label]
+  (if-let [source (gt/node-by-id-at basis source-id)] ; nil if source node was deleted in this transaction
+    (if-let [target (gt/node-by-id-at basis target-id)] ; nil if target node was deleted in this transaction
       (do
         (assert-type-compatible source-id source source-label target-id target target-label)
         (-> ctx
             ;; If the input has :one cardinality, disconnect existing connections first
             (ctx-disconnect-single target target-id target-label)
             (mark-input-activated target-id target-label)
             (update :basis gt/connect source-id source-label target-id target-label)
-            (flag-successors-changed [[source-id source-label]])
+            ;; When updating the successors, we must also consider any override
+            ;; nodes of the source node, since these will inherit an implicit
+            ;; connection between them and the corresponding override nodes of
+            ;; the target node.
+            (flag-successors-changed (cons (pair source-id source-label)
+                                           (e/map #(pair % source-label)
+                                                  (ig/get-overrides basis source-id))))
```

Comment:
Note: We use the "before" `basis` here since we want to consider the state before the change. Besides, override nodes are not created or destroyed immediately here, we just flag them for consideration at the end of the transaction.

Reply:
It seems that using `cons` erases all the benefits of using eduction here. Should we use `concat` with a singleton pair here instead? Or perhaps we could implement `cons` as a transducer...